src/action.js

```diff
@@ -55,10 +56,27 @@ async function run() {
     `);
 
     const pullRequest = data?.repository?.pullRequest;
-    const linkedIssuesCount = pullRequest?.closingIssuesReferences?.totalCount;
-    const issues = (pullRequest?.closingIssuesReferences?.nodes || []).map(
-      (node) => `${node.repository.nameWithOwner}#${node.number}`
-    );
+    let linkedIssuesCount = 0;
+    let issues = [];
+
+    const useLooseMatching = core.getBooleanInput("loose-matching", {
+      required: false,
+    });
+
+    if (useLooseMatching) {
+      issues = await getBodyValidIssue({
+        body: pullRequest.body,
+        repoName: name,
+        repoOwner: owner.login,
+        octokit,
+      });
+      linkedIssuesCount = issues.length;
+    } else {
+      linkedIssuesCount = pullRequest?.closingIssuesReferences?.totalCount;
+      issues = (pullRequest?.closingIssuesReferences?.nodes || []).map(
+        (node) => `${node.repository.nameWithOwner}#${node.number}`
+      );
+    }
```

Comment:
let's extract this in a function to avoid cluttering the main logic with low level details

Reply:
done